homeassistant/components/number/__init__.py

```diff
@@ -48,12 +48,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     component.async_register_entity_service(
         SERVICE_SET_VALUE,
         {vol.Required(ATTR_VALUE): vol.Coerce(float)},
-        "async_set_value",
+        async_set_value,
     )
 
     return True
 
 
+async def async_set_value(entry: NumberEntity, service_call: ServiceCall) -> None:
+    """Service call wrapper to set a new value."""
+    value = service_call.data["value"]
+    if value < entry.min_value or value > entry.max_value:
+        raise ValueError(
+            f"Value {value} not a valid for {entry.name} within the range {entry.min_value} - {entry.max_value}"
+        )
+    await entry.async_set_value(value)
```

Comment:
Just noticed this one, it's an entity, not an entry (which we generally use for config entries).

```suggestion
async def async_set_value(entity: NumberEntity, service_call: ServiceCall) -> None:
    """Service call wrapper to set a new value."""
    value = service_call.data["value"]
    if value < entity.min_value or value > entity.max_value:
        raise ValueError(
            f"Value {value} not a valid for {entity.name} within the range {entity.min_value} - {entity.max_value}"
        )
    await entity.async_set_value(value)
```